guvnor-m2repo-editor/guvnor-m2repo-editor-client/src/main/java/org/guvnor/m2repo/client/upload/UploadFormViewImpl.java

```diff
@@ -62,23 +59,28 @@
     private final TextBox hiddenArtifactIdField = new TextBox();
     private final TextBox hiddenVersionIdField = new TextBox();
 
+    private Presenter presenter;
+
     private final FormStyleLayout hiddenFieldsPanel = new FormStyleLayout();
 
+    private FileUpload uploader;
+
     @PostConstruct
     public void init() {
         popup.add( new ModalFooter( new Button( M2RepoEditorConstants.INSTANCE.Cancel() ) {{
             addClickHandler( new ClickHandler() {
                 @Override
                 public void onClick( ClickEvent event ) {
-                    hide();
+                    hideDialog();
```

Comment:
Personally, since we preserve use of the View's super-class ```show()``` I think it's more consistent to override ```hide()``` and ```hide(boolean autoClosed)``` than introduce a ```hideDialog()``` method. Also, but you wasn't to know this, it'd be best if we extend ```BaseModel``` than ```PopupPanel``` (as the former provides consistent behaviour of all of Business Central's "popups").